rectools/metrics/popularity.py

```diff
@@ -0,0 +1,174 @@
+    where "popularity" of item is the average number of ratings (interactions)
+    for this item.
+
+    .. math::
+        ARP@k = \frac{1}{\left|U_{t}\right|}\sum_{u\in U_{t}^{}}\frac{\sum_{i\in L_{u}}\phi (i)}{\left| L_{u} \right |}
+
+    where
+    :math:`\phi (i)` is the number of times item i has been rated in the training set.
+    :math:`|U_{t}|` is the number of users in the test set.
+    :math:`L_{u}` is the list of recommended items for user u.
+
+    Parameters
+    ----------
+    k : int
+        Number of items at the top of recommendations list that will be used to calculate metric.
+
+    Examples
+    --------
+    >>> reco = pd.DataFrame(
+    ...     {
+    ...         Columns.User: [1, 1, 2, 2, 2, 3, 3],
+    ...         Columns.Item: [1, 2, 3, 1, 2, 3, 2],
+    ...         Columns.Rank: [1, 2, 1, 2, 3, 1, 2],
+    ...     }
+    ... )
+    >>> prev_interactions = pd.DataFrame(
+    ...     {
+    ...         Columns.User: [1, 1, 2, 2, 3, 3],
+    ...         Columns.Item: [1, 2, 1, 3, 1, 2],
+    ...     }
+    ... )
+    >>> AvgRecPopularity(k=1).calc_per_user(reco, prev_interactions).values
+    array([3., 1., 1.])
+    >>> AvgRecPopularity(k=3).calc_per_user(reco, prev_interactions).values
+    array([2.5, 2. , 1.5])
+    """
+
+    def calc(self, reco: pd.DataFrame, prev_interactions: pd.DataFrame) -> float:
+        """
+        Calculate metric value.
+
+        Parameters
+        ----------
+        reco : pd.DataFrame
+            Recommendations table with columns `Columns.User`, `Columns.Item`, `Columns.Rank`.
+        prev_interactions : pd.DataFrame
+            Table with previous user-item interactions,
+            with columns `Columns.User`, `Columns.Item`.
+
+        Returns
+        -------
+        float
+            Value of metric (average between users).
+        """
+        per_user = self.calc_per_user(reco, prev_interactions)
+        return per_user.mean()
+
+    def calc_per_user(
+        self,
+        reco: pd.DataFrame,
+        prev_interactions: pd.DataFrame,
+    ) -> pd.Series:
+        """
+        Calculate metric values for all users.
+
+        Parameters
+        ----------
+        reco : pd.DataFrame
+            Recommendations table with columns `Columns.User`, `Columns.Item`, `Columns.Rank`.
+        prev_interactions : pd.DataFrame
+            Table with previous user-item interactions,
+            with columns `Columns.User`, `Columns.Item`.
+
+        Returns
+        -------
+        pd.Series
+            Values of metric (index - user id, values - metric value for every user).
+        """
+        pop_items = self.get_pop(prev_interactions)
+        reco_prepared = reco.query(f"{Columns.Rank} <= @self.k")
+        arp = reco_prepared.groupby(Columns.User)[Columns.Item].agg(
+            lambda x: sum(pop_items[i] if i in pop_items else 0 for i in x) / x.nunique()
+        )
+        return arp
+
+    def get_pop(self, prev_interactions: pd.DataFrame) -> pd.Series:
```

Comment:
This method does only one thing. So there is no need for a separate func. Just do value_counts() in the main func.

Reply:
`get_pop` method was deleted.